src/gg.regression.unity.bots/Runtime/Scripts/RemoteOrchestration/RGRemoteWorkerAssignmentManager.cs

```diff
@@ -0,0 +1,364 @@
+                            {
+                                clientId = this.ClientId,
+                                activeSequence = GetActiveBotSequence(),
+                                activeWorkAssignment = currentWorkAssignment
+                            };
+                            // send ACK of CANCEL immediately
+                            _ = RGServiceManager.GetInstance().SendRemoteWorkerHeartbeat(heartbeatRequest, HeartbeatResponseHandler, () => { });
+                        }
+                        else if (incomingWorkAssignment.status == WorkAssignmentStatus.COMPLETE_ERROR || incomingWorkAssignment.status == WorkAssignmentStatus.COMPLETE_SUCCESS || incomingWorkAssignment.status == WorkAssignmentStatus.COMPLETE_TIMEOUT)
+                        {
+                            // server ack'd our completion.. clear out the assignment
+                            var currentWorkAssignment = ActiveWorkAssignment;
+                            ActiveWorkAssignment = null;
+                            currentWorkAssignment.Stop();
+                            // immediately send another heartbeat in case there is a new assignment
+                            SendHeartbeatForCurrentState();
+                        }
+                        // else - we're still on the right assignment, just keep going if necessary
+                    }
+                }
+                else // ActiveWorkAssignment == null
+                {
+                    // check to make sure no other segment is running right now
+                    var activeSequence = GetActiveBotSequence();
+                    if (activeSequence != null)
+                    {
+                        // conflict - send a heartbeat for that conflict (don't update the time though so the regular ones keep flowing)
+                        incomingWorkAssignment.status = WorkAssignmentStatus.CONFLICT;
+                        incomingWorkAssignment.details = new ConflictDetails(null, incomingWorkAssignment.id);
+                        var heartbeatRequest = new SDKClientHeartbeatRequest()
+                        {
+                            clientId = this.ClientId,
+                            activeSequence = GetActiveBotSequence(),
+                            activeWorkAssignment = incomingWorkAssignment
+                        };
+                        // send report of CONFLICT immediately, don't mess with the active work assignment
+                        _ = RGServiceManager.GetInstance().SendRemoteWorkerHeartbeat(heartbeatRequest, HeartbeatResponseHandler, () => { });
+                    }
+                    else
+                    {
+                        // nothing else is running currently... GOOD... we mark this work assignment as ready to start on next update
+                        ActiveWorkAssignment = incomingWorkAssignment;
+                        ActiveWorkAssignment.status = WorkAssignmentStatus.WAITING_TO_START;
```

Comment:
I've been assuming the server will set the status as "WAITING_TO_START" at the time a work assignment is scheduled. This seems redundant, which doesn't hurt anything, but want to confirm whether an additional status like "SCHEDULED" would be useful